internal/subscriber/retry/delayconsumer_test.go

```diff
@@ -81,29 +116,148 @@ func TestDelayConsumer_Run(t *testing.T) {
 	dc.Run(ctx)
 }
 
-func getDummyBrokerMessage() messagebroker.ReceivedMessage {
-	tNow := time.Now()
-	tPast := tNow.Add(time.Second * 100 * -1)
+func TestDelayConsumer_Run_Consume(t *testing.T) {
+	ctx, ctrl, subCtx, cancel, brokerStore, handler, consumer, cache := setupDC(t)
+	subs := getMockSubscriptionModel()
+	subscriberID := "subscriber-id"
+	errCh := make(chan error)
+	dc, err := NewDelayConsumer(subCtx, subscriberID, "t1", subs, brokerStore, handler, cache, errCh)
+	producer := getMockProducer(ctx, ctrl)
+	brokerStore.EXPECT().GetProducer(gomock.Any(), gomock.Any()).Return(producer, nil).AnyTimes()
+
+	assert.NotNil(t, dc)
+	assert.Nil(t, err)
+	assert.NotNil(t, dc.LogFields())
+
+	messages := make([]messagebroker.ReceivedMessage, 0)
+	// message with delay within duration of test-case
+	msg1 := getDummyBrokerMessage("msg-1", "msg-id-1", time.Now().Add(time.Nanosecond*1000), 5)
+	// message with delay outside duration of test-case
+	msg2 := getDummyBrokerMessage("msg-2", "msg-id-2", time.Now().Add(time.Second*2), 5)
+	messages = append(messages, msg1, msg2)
+	count := 0
+
+	consumer.EXPECT().ReceiveMessages(gomock.AssignableToTypeOf(subCtx), gomock.Any()).DoAndReturn(
+		func(arg0 context.Context, arg1 messagebroker.GetMessagesFromTopicRequest) (*messagebroker.GetMessagesFromTopicResponse, error) {
+			if count == 0 {
+				count++
+				return &messagebroker.GetMessagesFromTopicResponse{Messages: messages}, nil
+			} else {
+				return &messagebroker.GetMessagesFromTopicResponse{}, nil
+			}
+		}).AnyTimes()
```

Comment:
can we move this func to setupDC func..?

Reply:
No since the messages will be different for different test cases

Reply:
You can pass msgs as parameter right..?

Reply:
@sakshi-goyal-razorpay setup is meant to set basic requirement for test cases, messages are not basic requirements this can be changed according to test cases, so why move these into setup?

Reply:
messages should be passed as input here based on the test case. If we go back to add more test cases in the future, this piece will require refactoring.

Reply:
message passed as input in test case

Reply:
@sumitkrzp This is still not addressed I guess

Reply:
```messages should be passed as input here based on the test case. If we go back to add more test cases in the future, this piece will require refactoring.``` in have addressed this, since putting recieveMessages expect call in the setup doesn't make sense, commented reasoning above